src/taxonomy/import-tags/data/api.js

```diff
@@ -0,0 +1,114 @@
+// @ts-check
+import { camelCaseObject, getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { useQueryClient, useMutation } from '@tanstack/react-query';
+
+const getApiBaseUrl = () => getConfig().STUDIO_BASE_URL;
+
+export const getTaxonomyImportNewApiUrl = () => new URL(
+  'api/content_tagging/v1/taxonomies/import/',
+  getApiBaseUrl(),
+).href;
+
+/**
+ * @param {number} taxonomyId
+ * @returns {string}
+ */
+export const getTagsImportApiUrl = (taxonomyId) => new URL(
+  `api/content_tagging/v1/taxonomies/${taxonomyId}/tags/import/`,
+  getApiBaseUrl(),
+).href;
+
+/**
+ * @param {number} taxonomyId
+ * @returns {string}
+ */
+export const getTagsPlanImportApiUrl = (taxonomyId) => new URL(
+  `api/content_tagging/v1/taxonomies/${taxonomyId}/tags/import/plan/`,
+  getApiBaseUrl(),
+).href;
+
+/**
+ * Import a new taxonomy
+ * @param {string} taxonomyName
+ * @param {string} taxonomyDescription
+ * @param {File} file
+ * @returns {Promise<Object>}
+ */
+export async function importNewTaxonomy(taxonomyName, taxonomyDescription, file) {
+  const formData = new FormData();
+  formData.append('taxonomy_name', taxonomyName);
+  formData.append('taxonomy_description', taxonomyDescription);
+  formData.append('file', file);
+
+  const { data } = await getAuthenticatedHttpClient().post(
+    getTaxonomyImportNewApiUrl(),
+    formData,
+  );
+
+  return camelCaseObject(data);
+}
+
+/**
+ * Build the mutation to import tags to an existing taxonomy
+ * @returns {import("@tanstack/react-query").UseMutationResult}
```

Comment:
```suggestion
```
Delete this line and the `@returns` type will be automatically (and correctly) inferred.